ApplicationCommands/src/Program.cs

```diff
@@ -0,0 +1,133 @@
+
+                // Register event handlers
+                appCommandShardExtension.SlashCommandExecuted += Slash_SlashCommandExecuted;
+                appCommandShardExtension.SlashCommandErrored += Slash_SlashCommandErrored;
+                appCommandShardExtension.ContextMenuExecuted += Context_ContextMenuCommandExecuted;
+                appCommandShardExtension.ContextMenuErrored += Context_ContextMenuCommandErrored;
+                
+                appCommandShardExtension.RegisterCommands<Ping>(guildId);
+                appCommandShardExtension.RegisterCommands<RoleInfo>(guildId);
+                appCommandShardExtension.RegisterCommands<RollRandom>(guildId);
+                appCommandShardExtension.RegisterCommands<Tags>(guildId);
+                appCommandShardExtension.RegisterCommands<Tell>(guildId);
+                appCommandShardExtension.RegisterCommands<TriggerHelp>(guildId);
+
+                // Currently, adding permissions to global commands during registration is not implemented
+                if (guildId != null)
+                {
+                    appCommandShardExtension.RegisterCommands<Slap>((ulong) guildId, context =>
+                    {
+                        // Allow members with the specified role from the arguments to execute the command
+                        if (args.Length > 2)
+                            context.AddRole(ulong.Parse(args[2]), true);
+                        
+                        // Allow owners of the bot to execute the command
+                        foreach (DiscordUser user in discordClient.CurrentApplication.Owners)
+                        {
+                            context.AddUser(user.Id, true);
+                        }
+                    });
+                    appCommandShardExtension.RegisterCommands<ManagePermissions>((ulong) guildId, context =>
+                    {
+                        // Allow owners of the bot to execute the command
+                        foreach (DiscordUser user in discordClient.CurrentApplication.Owners)
+                        {
+                            context.AddUser(user.Id, true);
+                        }
+                    });
+                }
+
+                // Context menu commands
+                appCommandShardExtension.RegisterCommands<MessageCopy>();
+                appCommandShardExtension.RegisterCommands<UserInfo>();
+            }
+            
+            discordShardedClient.Logger.LogInformation("Application commands registered successfully");
+
+            // Listen for commands by putting this method to sleep and relying off of DiscordClient's event listeners
+            await Task.Delay(-1);
+        }
+        
+        private static Task Slash_SlashCommandExecuted(ApplicationCommandsExtension sender, SlashCommandExecutedEventArgs e)
+        {
+            Console.WriteLine($"Slash/Info: {e.Context.CommandName}");
```

Comment:
Better logging formatting please. I recommend Serilog, however simply splitting everything by tabbed newlines would be better.